paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -198,7 +199,12 @@ public List<String> listTables(String databaseName) throws DatabaseNotExistExcep
             throw new DatabaseNotExistException(databaseName);
         }
 
-        return listTablesImpl(databaseName).stream().sorted().collect(Collectors.toList());
+        List<String> tables =
+                listTablesImpl(databaseName).stream().sorted().collect(Collectors.toList());
+        if (formatTableEnabled()) {
+            tables.addAll(listFormatTables(databaseName));
```

Comment:
Considering format table and normal paimon table are both tables, in most implementation they should be stored together, I prefer providing only one interface to list all tables no matter what kind of tables they are.